packages/nouns-webapp/src/hooks/useIdeas.ts

```diff
@@ -34,15 +34,24 @@ export const useIdeas = () => {
   const [ideas, setIdeas] = useState([] as Idea[]);
 
   // Update the vote count for an idea after a new vote is recorded.
-  const updateVotesState = ({ id, ideaId, direction }: Vote) => {
+  const updateVotesState = (vote: Vote) => {
+    const { id, ideaId, direction } = vote;
     const updatedIdeas = ideas.map(idea => {
       if (idea.id === ideaId) {
+        let seenVote = false;
         const newIdeaVotes = idea.votes.map(vote => {
           if (id === vote.id) {
+            seenVote = true;
             return { ...vote, direction };
+          } else {
+            return vote;
           }
         });
 
+        if (!seenVote) {
+          newIdeaVotes.push(vote);
```

Comment:
Ahhh is this because i was updating when a user changes their existing vote but wasn't handling the case when they vote for the first time? Good catch, i'd already voted on everything locally haha